src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,343 @@
+    return tile_url.format(image_base=image_base, zoom=zoom, depth=depth, row=row, col=col,
+                           file_extension=file_extension, tile_width=tile_width, tile_height=tile_height)
+
+
+def response_to_array(response):
+    """
+    response (Response): http response for the requested image.
+    return:
+       image (numpy array): the requested image from the response.
+    """
+    response.raise_for_status()
+    content_type = response.headers['Content-Type']
+
+    if content_type in SUPPORTED_CONTENT_TYPES:
+        buffer = BytesIO(response.content)  # opening directly from raw response doesn't work for JPEGs
+        raw_img = Image.open(buffer)
+        return numpy.array(raw_img)
+    else:
+        logging.info('Image fetching is only implemented for greyscale PNG and JPEG, not {}'.format(
+            content_type.upper().split('/')[1]
+        ))
+
+
+STACK_URL = "{base_url}/{project_id}/stack/{stack_id}/info"
+
+
+def get_stack_info(base_url, project_id, stack_id):
+    """Get the stack info for a project hosted at the base url."""
+    token, _, _ = read_config_file(base_url, token=True)
+    stack_url = STACK_URL.format(base_url=base_url, project_id=project_id, stack_id=stack_id)
+    response = requests.get(stack_url, auth=CatmaidApiTokenAuth(token)) if token else requests.get(stack_url)
+    # A PermissionError is raised when authentication fails.
+    if response.status_code == 404:
+        raise ValueError("no stack info at this url, check the base url {base_url}".format(base_url=base_url))
+    elif "PermissionError" in response.text:
+        raise AuthenticationError("Failed authentication getting stack info at {stack_url}".format(stack_url=stack_url))
+    elif "DoesNotExist" in response.text:
+        # Catmaid returns a 200 code with an error text field, when there's no stack info at that url.
+        raise ValueError(
+            "No stack info at this url {stack_url}, check project id {project_id} and stack id {stack_id}".format(
+                stack_url=stack_url, project_id=project_id, stack_id=stack_id))
+    stack_info = response.json()
+    return stack_info
+
+
+class CatmaidApiTokenAuth(HTTPBasicAuth):
+    """Attaches HTTP X-Authorization Token headers to the given Request.
+    Optionally, Basic HTTP Authentication can be used in parallel.
+    """
+
+    def __init__(self, token, username=None, password=None):
+        super(CatmaidApiTokenAuth, self).__init__(username, password)
+        self.token = token
+
+    def __call__(self, r):
+        r.headers['X-Authorization'] = 'Token {}'.format(self.token)
+        if self.username and self.password:
+            super(CatmaidApiTokenAuth, self).__call__(r)
+        return r
+
+
+def read_config_file(base_url, token=False, username=False, password=False):
+    """
+    Read a config file and return the token, username and password corresponding to the given base_url.
```

Comment:
Please document the format at the top of this python file (and indicating the path of the config file, and how to get a token in Catmaid). Ideally, put a small example of how the file looks like too.